core/store/src/test/java/uk/gov/gchq/gaffer/store/library/AbstractGraphLibraryTest.java

```diff
@@ -0,0 +1,278 @@
+        graphLibrary.clear();
+    }
+
+    @Test
+    public void shouldClearGraphLibrary() {
+        // When
+        graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties);
+        graphLibrary.clear();
+
+        // Then
+        assertEquals(null, graphLibrary.getIds(TEST_GRAPH_ID));
+        assertEquals(null, graphLibrary.getSchema(TEST_SCHEMA_ID));
+        assertEquals(null, graphLibrary.getProperties(TEST_PROPERTIES_ID));
+
+    }
+
+    @Test
+    public void shouldAddAndGetMultipleIdsInGraphLibrary() {
+        // When
+        graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties);
+        graphLibrary.add(TEST_GRAPH_ID_1, schema1, storeProperties1);
+    }
+
+    @Test
+    public void shouldAddAndGetIdsInGraphLibrary() {
+        // When
+        graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties);
+
+        // Then
+        assertEquals(new Pair<>(TEST_SCHEMA_ID, TEST_PROPERTIES_ID), graphLibrary.getIds(TEST_GRAPH_ID));
+    }
+
+    @Test
+    public void shouldThrowExceptionWithInvalidGraphId() {
+        // When / Then
+        try {
+            graphLibrary.add(TEST_GRAPH_ID + "@#", schema, storeProperties);
+            fail("Exception expected");
+        } catch (final IllegalArgumentException e) {
+            assertNotNull(e.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldAddAndGetSchema() {
+        // When
+        graphLibrary.addSchema(schema);
+
+        // Then
+        JsonAssert.assertEquals(schema.toJson(false), graphLibrary.getSchema(schema.getId()).toJson(false));
+    }
+
+    @Test
+    public void shouldNotAddNullSchema() {
+        // When / Then
+        try {
+            graphLibrary.addSchema(null);
+        } catch (final IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("Id is invalid: null"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenGraphIdWithDifferentSchemaExists() {
+
+        // Given
+        graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties);
+
+        // When / Then
+        try {
+            graphLibrary.add(TEST_GRAPH_ID, schema1, storeProperties);
+            fail("Exception expected");
+        } catch (final OverwritingException e) {
+            assertTrue(e.getMessage().contains("already exists with a different schema"));
+        }
+    }
+
+    @Test
+    public void shouldAddAndGetProperties() {
+        // When
+        graphLibrary.addProperties(storeProperties);
+
+        // Then
+        assertEquals(storeProperties, graphLibrary.getProperties(storeProperties.getId()));
+    }
+
+    @Test
+    public void shouldNotAddNullProperties() {
+        // When / Then
+        try {
+            graphLibrary.addProperties(null);
+        } catch (final IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("Id is invalid: null"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenGraphIdWithDifferentPropertiesExists() {
```

Comment:
add won for when schema and properties is the same. what is the expected behaviour

Reply:
This is already there